pkg/restore/restore.go

```diff
@@ -917,44 +939,57 @@ func (r *pvRestorer) executePVAction(obj *unstructured.Unstructured) (*unstructu
 		return obj, nil
 	}
 
-	// If we can't find a snapshot record for this particular PV, it most likely wasn't a PV that Ark
-	// could snapshot, so return early instead of trying to restore from a snapshot.
-	backupInfo, found := r.volumeBackups[pvName]
-	if !found {
-		return obj, nil
-	}
-
-	// Past this point, we expect to be doing a restore
+	for _, snapshot := range r.volumeSnapshots {
```

Comment:
let's invert this loop to remove nesting and handle the error case first - so just loop through `r.volumeSnapshots` looking for the snapshot with the right `PersistentVolumeName`.  If it's found, break.  After the loop, if we didn't find the snapshot, it shouldn't be an error (PVs aren't required to have snapshots -- see https://github.com/heptio/ark/pull/938/files#diff-7e2ddda55b692c82f9fa5f89a3d05e54L920), so just return.  Otherwise, if we did find a snapshot, then go through the rest of the logic that's currently inside the loop.